_posts/2023-09-20-quarkus-observability-3-3.adoc

```diff
@@ -0,0 +1,178 @@
+
+=== Exporter now supports the HTTP protocol
+
+Until Quarkus 3.2, the OTel exporter only allowed the default GRPC protocol. Now it also supports HTTP. To use the new HTTP protocol, the `quarkus.otel.exporter.otlp.traces.protocol` property must be set to `http/protobuf`. The `quarkus.otel.traces.exporter.endpoint` property must also be set to the OTel Collector HTTP endpoint. Here's an example when using 4318, the default HTTP port on the OTel Collector:
+```application.properties
+quarkus.otel.exporter.otlp.traces.protocol=http/protobuf
+quarkus.otel.exporter.otlp.endpoint=http://localhost:4318
+```
+
+=== Exporters support TLS
+
+Both GRPC and HTTP exporters now support Transport Layer Security (TLS) and custom certificates.
+
+Like the other rest clients in Quarkus, the exporter's certificate verification will also be disabled if you set:
+```application.properties
+quarkus.tls.trust-all=true
+```
+[WARNING]
+====
+This setting should not be used in production as it will disable any kind of SSL verification.
+====
+
+=== Customise the propagation header
+
+We added a way to customise the propagation header.
+
+You can create a CDI bean implementing the `TextMapPropagatorCustomizer` interface.
+
+This can be used to restrict propagation of OpenTelemetry trace headers and prevent potentially sensitive data to be sent to third party systems.
+
+This is an example of a customizer that removes the Baggage header:
+```java
+@Singleton
+public static class TestTextMapPropagatorCustomizer implements TextMapPropagatorCustomizer {
+    @Override
+    public TextMapPropagator customize(Context context) {
+        TextMapPropagator propagator = context.propagator();
+        if (propagator instanceof W3CBaggagePropagator) {
+            return TextMapPropagator.noop();
+        }
+        return propagator;
+    }
+}
+```
+
+=== Identify the user in the spans
+
+Valuable debugging information about the user related to each span can be added by setting:
+```application.properties
+quarkus.otel.traces.eusp.enabled=true
+```
+The user’s ID and roles will be added to the span attributes, if available.
+
+=== Polish
+
+Many bug fixes and small improvements were made to the extension, including:
+
+* Reduce the memory allocation needed to report spans.
+* Fix the `http.route` span attribute value in some cases.
+* Properly report of the `service.name` value. The value can be set in 3 different properties, from higher to lower priority:
+1. `quarkus.otel.service.name`
+2. `quarkus.otel.resource.attributes`
+3. `quarkus.application.name`
+* Exception's stack traces are now reported in the span attributes.
+
+=== OpenTelemetry upgrades.
+
+OTel moves fast! Since working on Quarkus 3.0 we did 7 version upgrades, moving `opentelemetry-java` from version 1.21.0 to 1.28.0.
```

Comment:
```suggestion
OTel moves fast! Since working on Quarkus 3.0 we have performed 7 version upgrades, moving `opentelemetry-java` from version 1.21.0 to 1.28.0.
```

Reply:
ok